runtime/src/main/java/org/corfudb/runtime/collections/SMRMultiIndex.java

```diff
@@ -0,0 +1,362 @@
+     */
+    @Accessor
+    public Collection<Map.Entry<K,P>> getByNamedIndex(String indexName, I indexKey) {
+        Map<I,Map<K,P>> indexMap = indexMaps.get(indexName);
+        if (indexMap == null)
+            return Collections.EMPTY_SET;
+        Map<K,P> index = indexMap.get(indexKey);
+        if (index == null)
+            return Collections.EMPTY_SET;
+        return index.entrySet();
+    }
+
+    /** Get the size of the multi-index.
+     * @return  The number of entries mapped (rows).
+     */
+    @Accessor
+    public int size() {
+        return mainMap.size();
+    }
+
+    @Accessor
+    @Override
+    public boolean isEmpty() {
+        return mainMap.isEmpty();
+    }
+
+    @Accessor
+    @Override
+    public boolean containsKey(Object key) {
+        return mainMap.containsKey(key);
+    }
+
+    @Accessor
+    @Override
+    public boolean containsValue(Object value) {
+        return mainMap.containsValue(value);
+    }
+
+
+
+    /** Clear the multi-index.
+     */
+    @Mutator(reset = true, name = "clear")
+    public void clear() {
+        throw new RuntimeException("###########Operation Not Implemented.##########");
+
+    }
+
+    @Accessor
+    @Override
+    public Set<K> keySet() {
+        return mainMap.keySet();
+    }
+
+    @Accessor
+    @Override
+    public Collection<V> values() {
+        return mainMap.values();
+    }
+
+    @Accessor
+    @Override
+    public Set<Entry<K, V>> entrySet() {
+        return mainMap.entrySet();
+    }
+
+
+    //--------------- Helper Methods to Create/Update/Remove Indexes -----------
+
+    public void createIndexes(K key, V value) {
+        for(Map.Entry<String, IndexSpecification> entry: indexSpecifications.entrySet()){
+            String indexName = entry.getKey();
+            IndexSpecification indexSpecification = entry.getValue();
+            List<Map<K, P>> indexList = getIndexList(indexName, indexSpecification, key, value);
+            indexList.forEach(index -> index.put(key, (P)indexSpecification.projectionFunction.generateProjection(key, value)));
+        }
+    }
+    public void removeIndexes(K key, V value) {
+        for(Map.Entry<String, IndexSpecification> entry: indexSpecifications.entrySet()){
+            String indexName = entry.getKey();
+            IndexSpecification indexSpecification = entry.getValue();
+            List<Map<K, P>> indexList = getIndexList(indexName, indexSpecification, key, value);
+            indexList.forEach(index -> index.remove(key, (P)indexSpecification.projectionFunction.generateProjection(key, value)));
+        }
+    }
+
+    public List<Map<K, P>> getIndexList(String indexName, IndexSpecification indexSpecification, K key, V value) {
+
+        List<Map<K, P>> indexes = new ArrayList<Map<K, P>>();
+
+        for (Object indexFunction : indexSpecification.getIndexFunctions()) {
+            I indexKey = (I) ((IndexFunction<K, V, I>) indexFunction).generateIndex(key, value);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "I". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1905)